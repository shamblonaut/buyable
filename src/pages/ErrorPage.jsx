import { useRouteError, Link } from "react-router-dom";
import styled from "styled-components";

const PageContainer = styled.main`
  height: 100svh;
  display: flex;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

const ErrorPage = () => {
  const error = useRouteError();

  return (
    <PageContainer>
      <h2>
        Error {error.status}: {error.statusText}
      </h2>
      <p>{error.data}</p>
      <Link to="/">Return to Home</Link>
    </PageContainer>
  );
};

export default ErrorPage;
