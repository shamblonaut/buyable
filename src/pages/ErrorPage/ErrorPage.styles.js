import styled from "styled-components";

export const Page = styled.main`
  height: 100svh;
  display: flex;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%;
`;
