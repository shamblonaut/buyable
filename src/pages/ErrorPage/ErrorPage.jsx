import { useRouteError, Link } from "react-router";

import { Page } from "./ErrorPage.styles";

const ErrorPage = () => {
  const error = useRouteError();

  return (
    <Page>
      <h2>
        Error {error.status}: {error.statusText}
      </h2>
      <p>{error.data}</p>
      <Link to="/">Return to Home</Link>
    </Page>
  );
};

export default ErrorPage;
