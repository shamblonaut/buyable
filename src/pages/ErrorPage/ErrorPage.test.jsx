import { describe, it, expect, vi, beforeEach } from "vitest";
import { render, screen } from "@testing-library/react";

import { MemoryRouter } from "react-router";

import ErrorPage from "./ErrorPage";

const mockError = {
  status: 404,
  statusText: "Not Found",
  data: "No route matches URL",
};

vi.mock("react-router", async (importOriginal) => {
  const actual = await importOriginal();
  return {
    ...actual,
    useRouteError: () => mockError,
  };
});

describe("Error Page", () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  it("renders correctly", () => {
    const { container } = render(
      <MemoryRouter>
        <ErrorPage />
      </MemoryRouter>,
    );

    expect(container).toMatchSnapshot();
  });

  it("displays all error information", () => {
    render(
      <MemoryRouter>
        <ErrorPage />
      </MemoryRouter>,
    );

    expect(
      screen.getByText(mockError.status, { exact: false }),
    ).toBeInTheDocument();
    expect(
      screen.getByText(mockError.statusText, { exact: false }),
    ).toBeInTheDocument();
    expect(
      screen.getByText(mockError.data, { exact: false }),
    ).toBeInTheDocument();
  });

  it("links to the home page", () => {
    render(
      <MemoryRouter>
        <ErrorPage />
      </MemoryRouter>,
    );

    expect(screen.getByRole("link")).toHaveAttribute("href", "/");
  });
});
