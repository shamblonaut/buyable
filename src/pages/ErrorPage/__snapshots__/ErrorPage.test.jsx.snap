// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Error Page > renders correctly 1`] = `
<div>
  <main
    class="sc-bRKDuR dSkNew"
  >
    <h2>
      Error 
      404
      : 
      Not Found
    </h2>
    <p>
      No route matches URL
    </p>
    <a
      data-discover="true"
      href="/"
    >
      Return to Home
    </a>
  </main>
</div>
`;
